### ISSUE TITLE
[Sourcing-2][Accessibility: ACC-253.1][WIP] The search field, within Table does not have visible label defining the purpose.

### ISSUE THREAD

**Issue by JyotiKumari1507** [SAP/fundamental-ngx#11241]
#### Is this a bug, enhancement, or feature request?
Bug
#### Describe your proposal.
Search label in Platform Table should be customisable by user. As per accessibility Team bases on what kind of Table field we are searching its should be that not by default 'Search' always. Like for Employee table we should show 'Search for Employee'

#### Can you handle that on the application side
No

#### Which versions of Angular and Fundamental Library for Angular are affected? Please, specify the exact version. (If this is a feature request, use current version.)
All latest version

#### If this is a bug, please provide steps for reproducing it; the exact components you are using;
NGX Platform Table toolbar menu search textbox

#### In case this is Accessibility related topic, did you consult with an accessibility expert? If not, please do so and share their recommendations. 
Yes we have got this bug from Accessibility expert only.

#### Did you check the documentation and the API?
Yes
#### Did you search for similar issues?
Yes
#### Is there anything else we should know?
Attaching screenshot for references
<img width="1235" alt="Screenshot 2024-01-21 at 11 20 50 AM" src="https://github.com/SAP/fundamental-ngx/assets/157268666/6d7d4647-d6fd-4c27-b362-2b646d6b70b0">

**Comment by droshev** [comment]
@JyotiKumari1507 you can customize the toolbar and the content of it with core table like this [example](https://fundamental-ngx.netlify.app/#/core/table#table-toolbar)

@nikolay-kolarov do you think we have to offer more than that?

**Comment by JyotiKumari1507** [comment]
> @JyotiKumari1507 you can customize the toolbar and the content of it with core table like this [example](https://fundamental-ngx.netlify.app/#/core/table#table-toolbar)
> 
> @nikolay-kolarov do you think we have to offer more than that?

@droshev Accessibility Team expecting Placeholder label of Search should be more customise where user can decide based on Table. Currently its fixed 'Search'. We can discussed about this more in next week call.

**Comment by nikolay-kolarov** [comment]
Hi @droshev ,

In the context of the SearchField the placeholder, together with the search icon give enough context that this is a search field and as an exception, a real label in front can be skipped (this is approved by UX and ACC).

Best Regards,
Nikolay

**Comment by nikolay-kolarov** [comment]
Going through the report once again, I see that the focus is more on the placeholder and not on external label.
I think the reporter needs an API to check the placeholder from application side, which is a feature request. In UI5 such an API exisit, so I do not see issues to introduce it here as well.

Best Regards,
Nikolay

**Comment by droshev** [comment]
You can control the search field if you use Core table. there you have full control of all attributes and inputs.

**Comment by JyotiKumari1507** [comment]
@droshev pls let me know the NGX version where this fix available?

**Comment by droshev** [comment]
@JyotiKumari1507 there wasn't a fix. you can check the [API](https://fundamental-ngx.netlify.app/#/platform/search-field/api) for everything you need

**Comment by divyamalhotra488** [comment]
Hi @droshev, we are not using the search component directly. That is coming as part of platform table component.
We don't have control of Search label over there.
Based on accessibility team suggestion, either the label should be able to customise or we should have tooltip on the Search icon.

**Comment by droshev** [comment]
@divyamalhotra488 the component `TableToolbarComponent` offers an input `searchFieldAriaLabel` which you can use to pass what you need I think
